warehouse/warehouse.go

```diff
@@ -725,6 +729,59 @@ func (wh *HandleT) runUploadJobAllocator() {
 	}
 }
 
+func (wh *HandleT) uploadStatusTrack() {
+	for {
+		time.Sleep(uploadStatusTrackFrequency)
+
+		for _, warehouse := range wh.warehouses {
+			source := warehouse.Source
+			destination := warehouse.Destination
+
+			if !source.Enabled || !destination.Enabled {
+				continue
+			}
+
+			config := destination.Config
+			// Default frequency
+			syncFrequency := "1440"
+			if config[warehouseutils.SyncFrequency] != nil {
+				syncFrequency, _ = config[warehouseutils.SyncFrequency].(string)
+			}
+
+			timeWindow := uploadBufferTimeInMin
+			if value, err := strconv.Atoi(syncFrequency); err == nil {
+				timeWindow += value
+			}
+
+			sqlStatement := fmt.Sprintf(`
+				select 
+					cast( case when count(*) > 0 then 1 else 0 end as bit ), 
+						(select count(*) from %[7]s where source_id='%[2]s' and destination_id='%[3]s' and created_at > now() - interval '%[6]d MIN') as staging_files_count 
+				from %[1]s where source_id='%[2]s' and destination_id='%[3]s' and (status='%[4]s' or status='%[5]s') and updated_at > now() - interval '%[6]d MIN'`,
+				warehouseutils.WarehouseUploadsTable, source.ID, destination.ID, ExportedData, Aborted, timeWindow, warehouseutils.WarehouseStagingFilesTable)
+
+			rows, err := wh.dbHandle.Query(sqlStatement)
```

Comment:
make it `QueryRow` since only 1 row is returned from query always

Reply:
done